mir/src/passes/translate.rs

```diff
@@ -1281,10 +1285,10 @@ impl<'a> MirBuilder<'a> {
     // Check assumptions, probably this assumed that the inlining pass did some work
     fn trace_access(&self, access: &ast::SymbolAccess) -> Option<TraceAccess> {
         let id = access.name.as_ref();
-        for (i, segment) in self.trace_columns.iter().enumerate() {
+        if let Some(segment) = self.trace_columns.first() {
```

Comment:
Should we add an `assert!` to make sure that `self.trace_columns` has only one segment?